model.py

```diff
@@ -3448,12 +3400,9 @@ def calculate_presentation(self, policy=None, search_index_client=None):
             # classifications, or measurements.
             _db = Session.object_session(self)
 
-            primary_identifier_ids = [
-                x.primary_identifier.id for x in self.editions
-                if x.primary_identifier
-            ]
+            primary_identifier_id = self.presentation_edition.primary_identifier.id
```

Comment:
Once again, I think you want all the identifiers associated with all the LicensePools.